test/e2e/imagebundle/serve_bundle_oci_test.go

```diff
@@ -0,0 +1,222 @@
+			"--listen-port", strconv.Itoa(port),
+			"--tls-cert-file", certFile,
+			"--tls-private-key-file", keyFile,
+			"--repositories-prefix", "/some/test/prefix",
+		})
+
+		done := make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+
+			Expect(cmd.Execute()).To(Succeed())
+
+			close(done)
+		}()
+
+		helpers.WaitForTCPPort(GinkgoT(), ipAddr.String(), port)
+
+		testRoundTripper, err := httputils.TLSConfiguredRoundTripper(
+			remote.DefaultTransport,
+			net.JoinHostPort(ipAddr.String(), strconv.Itoa(port)),
+			false,
+			caCertFile,
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, imageRef := range expectedOCIArtifacts {
+			mindthegapRef := fmt.Sprintf("%s:%d%s/%s", ipAddr, port, "/some/test/prefix", imageRef)
+			ref, err := name.ParseReference(mindthegapRef)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = remote.Image(ref, remote.WithTransport(testRoundTripper))
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		close(stopCh)
+
+		Eventually(done).Should(BeClosed())
+	})
+
+	It("failes to create bundle with OCI image in OCI artifacts list", func() {
+		imagesTxt := filepath.Join(GinkgoT().TempDir(), "oic-images.txt")
+		Expect(os.WriteFile(imagesTxt, []byte("stefanprodan/podinfo:6.8.0"), 0644)).To(Succeed())
+		createBundleCmd := helpers.NewCommand(GinkgoT(), createbundle.NewCommand)
+		createBundleCmd.SetArgs([]string{
+			"--output-file", bundleFile,
+			"--oci-artifacts-file", imagesTxt,
+		})
+		Expect(createBundleCmd.Execute()).To(MatchError(ContainSubstring("unexpected media type in descriptor for OCI artifact")))
+	})
+
+	It("pushes OCI artifacts to the registry", func() {
+		helpers.CreateBundleOCI(
+			GinkgoT(),
+			bundleFile,
+			filepath.Join("testdata", "create-success-oci.yaml"),
+		)
+
+		port, err := freeport.GetFreePort()
+		Expect(err).NotTo(HaveOccurred())
+		reg, err := registry.NewRegistry(registry.Config{
+			Storage: registry.FilesystemStorage(filepath.Join(GinkgoT().TempDir(), "registry")),
+			Host:    "127.0.0.1",
+			Port:    uint16(port),
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		done := make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+
+			Expect(
+				reg.ListenAndServe(
+					funcr.New(func(prefix, args string) {
+						log.Println(prefix, args)
+					}, funcr.Options{}),
+				),
+			).To(Succeed())
+
+			close(done)
+		}()
+
+		helpers.WaitForTCPPort(GinkgoT(), "127.0.0.1", port)
+
+		cmd := helpers.NewCommand(
+			GinkgoT(),
+			func(out output.Output) *cobra.Command { return pushbundle.NewCommand(out, "bundle") },
+		)
+		args := []string{
+			"--bundle", bundleFile,
+			"--to-registry", fmt.Sprintf("127.0.0.1:%d", port),
+		}
+		cmd.SetArgs(args)
+		Expect(cmd.Execute()).To(Succeed())
+
+		for _, imageRef := range expectedOCIArtifacts {
+			mindthegapRef := fmt.Sprintf("127.0.0.1:%d/%s", port, imageRef)
+			ref, err := name.ParseReference(mindthegapRef)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = remote.Image(ref)
+			Expect(err).NotTo(HaveOccurred())
+		}
```

Comment:
While this works, it has broken pushing images and OCI artifacts right now, tested locally.

We will need to update `push/bundle/bundle.go` too here to handle both images config files.

Reply:
Tested locally and this is definitely broken btw - please add an e2e that combines both OCI and images configs.

Reply:
Thanks, I've restructured the code and use only single `images.yaml` file for both images and artifacts. Is that OK? I can easily restructure this to use a dedicated file for artifacts.

Reply:
LGTM!